DanTup.BrowserSelector/Program.cs

```diff
@@ -1,60 +1,124 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Reflection;
 using System.Security.Principal;
+using System.Text;
 using System.Text.RegularExpressions;
 using System.Windows.Forms;
+using System.Xml;
 
 namespace DanTup.BrowserSelector
 {
 	class Program
 	{
-        static readonly Dictionary<string, string> SpecialCommands = new Dictionary<string, string>
-	    {
-	        {"edge", "microsoft-edge:{0}"}
-	    };
-
-        static void Main(string[] args)
+		static void Main(string[] args)
 		{
-			if (args == null || args.Length != 1 || !HandleArg(args[0]))
-				ShowHelpInfo();
-		}
+			string arg;
+			bool isOption,
+				waitForClose = false;
 
-		static bool HandleArg(string arg)
-		{
-			if (string.Equals(arg, "--register", StringComparison.OrdinalIgnoreCase))
+			if (args == null || args.Length == 0)
 			{
-				EnsureAdmin(arg);
-				RegistrySettings.RegisterBrowser();
+				ShowHelpInfo();
+				return;
 			}
-			else if (string.Equals(arg, "--unregister", StringComparison.OrdinalIgnoreCase))
+
+			for (int i = 0; i < args.Length; i++)
 			{
-				EnsureAdmin(arg);
-				RegistrySettings.UnregisterBrowser();
-			}
-			else if (arg.StartsWith("http://", StringComparison.OrdinalIgnoreCase) || arg.StartsWith("https://", StringComparison.OrdinalIgnoreCase) || arg.StartsWith("ftp://", StringComparison.OrdinalIgnoreCase))
-				LaunchBrowser(arg);
-			else
-				return false;
+				arg = args[i].Trim();
+
+				isOption = arg.StartsWith("-") || arg.StartsWith("/");
+				while (arg.StartsWith("-") || arg.StartsWith("/"))
+					arg = arg.Substring(1);
 
-			return true;
+				if (isOption)
+				{
+					if (string.Equals(arg, "register", StringComparison.OrdinalIgnoreCase))
+					{
+						EnsureAdmin(arg);
+						RegistrySettings.RegisterBrowser();
+						return;
+					}
+					else if (string.Equals(arg, "unregister", StringComparison.OrdinalIgnoreCase))
+					{
+						EnsureAdmin(arg);
+						RegistrySettings.UnregisterBrowser();
+						return;
+					}
+					else if (string.Equals(arg, "create", StringComparison.OrdinalIgnoreCase))
+					{
+						CreateSampleSettings();
+						return;
+					}
+					else if (string.Equals(arg, "wait", StringComparison.InvariantCultureIgnoreCase))
+					{
+						waitForClose = true;
+					}
+					else if (string.Equals(arg, "no-wait", StringComparison.InvariantCultureIgnoreCase))
+					{
+						waitForClose = false;
+					}
+				}
+				else
+				{
+					if (arg.EndsWith(".url", StringComparison.InvariantCultureIgnoreCase) || arg.EndsWith(".website", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
switched the order.. :)


Kody Brown
thewizard@wasatchwizard.com
http://wasatchwizard.com

On Thu, Feb 11, 2016 at 12:50 PM, Danny Tuppeny <notifications@github.com>
wrote:

> In DanTup.BrowserSelector/Program.cs
> <https://github.com/DanTup/BrowserSelector/pull/5#discussion_r52656687>:
>
> > +					{
> > +						CreateSampleSettings();
> > +						return;
> > +					}
> > +					else if (string.Equals(arg, "wait", StringComparison.InvariantCultureIgnoreCase))
> > +					{
> > +						waitForClose = true;
> > +					}
> > +					else if (string.Equals(arg, "no-wait", StringComparison.InvariantCultureIgnoreCase))
> > +					{
> > +						waitForClose = false;
> > +					}
> > +				}
> > +				else
> > +				{
> > +					if (arg.EndsWith(".url", StringComparison.InvariantCultureIgnoreCase) || arg.EndsWith(".website", StringComparison.InvariantCultureIgnoreCase))
>
> I think it'd be best to reverse the order of these, just in case anyone
> has a url that ends with .url, .website, etc. (sure, it's unlikely, but
> the other way there's no such issue; since starting with http(s)/ftp is
> absolutely clear).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DanTup/BrowserSelector/pull/5/files#r52656687>.
>